src/Uno.UI/UI/Xaml/Controls/ToolTip/ToolTip.cs

```diff
@@ -73,10 +129,600 @@ private void AttachToPopup()
+			}
+			else if (!placementRect.IsEmpty)
+			{
+				rcDockTo = placementRect;
+			}
+			else if (!m_isSliderThumbToolTip
+				// UNOTODO
+				//&&
+				//      (AutomaticToolTipInputMode::Touch == m_inputMode || AutomaticToolTipInputMode::Mouse == m_inputMode)
+				)
+			{
+				var lastPointerEnteredPoint = default(Point);
+
+				// UNO TODO: PointerPoint.GetCurrentPoint(uint pointerId)
+				lastPointerEnteredPoint = CoreWindow.GetForCurrentThread().PointerPosition;
+
+				rcDockTo.X = lastPointerEnteredPoint.X;
+				rcDockTo.Y = lastPointerEnteredPoint.Y;
+
+				// UNO TODO
+				//// For touch, we need to account for the fact that the context menu hint has a vertical offset.
+				//if (AutomaticToolTipInputMode::Touch == m_inputMode)
+				//{
+				//    rcDockTo.Top += CONTEXT_MENU_HINT_VERTICAL_OFFSET;
+				//    rcDockTo.bottom += CONTEXT_MENU_HINT_VERTICAL_OFFSET;
+				//}
+			}
+			else
+			{
+				getDockToRectFromTargetElement = true;
+			}
+
+			if (getDockToRectFromTargetElement)
+			{
+				var targetTopLeft = default(Point);
+				Target.TransformToVisual(null).TransformPoint(targetTopLeft);
+
+				//// UNO TODO
+				//      // targetTopLeft.X should be the left edge of the target, so adjust for RTL by
+				//      // subtracting the width of the target.
+				//      if (bIsRTL)
+				//      {
+				//          targetTopLeft.X -= targetActualWidth;
+				//      }
+
+				rcDockTo.X = targetTopLeft.X;
+				rcDockTo.Y = targetTopLeft.Y;
+				rcDockTo.Width = Target.ActualWidth;
+				rcDockTo.Height = Target.ActualHeight;
+			}
+
+			rcDockTo = rcDockTo.OffsetRect(origin.X, origin.Y);
+
+			// If horizontal & vertical offset are not specified, use the system defaults.
+			var isPropertyLocal = this.IsDependencyPropertySet(HorizontalOffsetProperty);
+			if (!isPropertyLocal)
+			{
+				// UNO TODO
+				//switch (m_inputMode)
+				//{
+				//case AutomaticToolTipInputMode::Keyboard:
+				//    horizontalOffset = DEFAULT_KEYBOARD_OFFSET;
+				//    break;
+				//case AutomaticToolTipInputMode::Mouse:
+				//    horizontalOffset = DEFAULT_MOUSE_OFFSET;
+				//    break;
+				//case AutomaticToolTipInputMode::Touch:
+				//    horizontalOffset = DEFAULT_TOUCH_OFFSET;
+				//    break;
+				//}
+				horizontalOffset = DEFAULT_MOUSE_OFFSET;
+			}
+
+			isPropertyLocal = this.IsDependencyPropertySet(VerticalOffsetProperty);
+			if (!isPropertyLocal)
+			{
+				// UNO TODO
+				//switch (m_inputMode)
+				//{
+				//case AutomaticToolTipInputMode::Keyboard:
+				//    verticalOffset = DEFAULT_KEYBOARD_OFFSET;
+				//    break;
+				//case AutomaticToolTipInputMode::Mouse:
+				//    verticalOffset = DEFAULT_MOUSE_OFFSET;
+				//    break;
+				//case AutomaticToolTipInputMode::Touch:
+				//    verticalOffset = DEFAULT_TOUCH_OFFSET;
+				//    break;
+				//}
+
+				verticalOffset = DEFAULT_MOUSE_OFFSET;
+			}
+
+			// UNO TODO
+			//// Reverse Left/Right palacement for RTL, because the target is flipped.
+			//if (bIsRTL)
+			//{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5404644)